lib/IRGen/DebugTypeInfo.h

```diff
@@ -100,7 +100,7 @@ class DebugTypeInfo {
   // instead? Otherwise optionals of archetypes etc will still have
   // 'isImplicitlyIndirect()' return false.
   bool isImplicitlyIndirect() const {
-    return Type->hasLValueType() || isArchetype() || Type->is<InOutType>();
+    return Type->hasLValueType() || isArchetype();
```

Comment:
When do you see LValue types in DebugInfo? They shouldn't really exist outside of type checking unless you're constructing them yourself with LValueType::get()